latch_cli/services/cp/autocomplete.py

```diff
@@ -0,0 +1,257 @@
+    children = _get_immediate_children_of_node(parent)
+
+    res: List[sc.CompletionItem] = []
+    for child in children:
+        if child.startswith(stub):
+            res.append(sc.CompletionItem(urljoins(parent, child)))
+
+    return res
+
+
+domain = re.compile(r"^(latch)?://(?P<stub>[^/]*)$")
+
+
+# `incomplete` assumed to be of the form '(latch)?://[^/]*'
+@cache
+def _complete_domain(incomplete: str) -> List[sc.CompletionItem]:
+    match = domain.match(incomplete)
+    if match is None:
+        return []
+
+    stub = match["stub"]
+
+    res: List[sc.CompletionItem] = []
+    for d in _get_all_valid_domains_for_account():
+        if d.startswith(stub):
+            res.append(sc.CompletionItem(f"latch://{d}/"))
+
+    return res
+
+
+class Child(TypedDict):
+    name: str
+
+
+class ChildLdataTreeEdgesNode(TypedDict):
+    child: Child
+
+
+class ChildLdataTreeEdges(TypedDict):
+    nodes: List[ChildLdataTreeEdgesNode]
+
+
+class LdataResolvePathData(TypedDict):
+    childLdataTreeEdges: ChildLdataTreeEdges
+
+
+@cache
+def _get_immediate_children_of_node(path: str) -> List[str]:
+    lrpd: LdataResolvePathData = execute(
+        gql.gql("""
+            query MyQuery($argPath: String!) {
+                ldataResolvePathData(argPath: $argPath) {
+                    childLdataTreeEdges(
+                        filter: {child: {removed: {equalTo: false}}}
+                    ) {
+                        nodes {
+                            child {
+                                name
+                            }
+                        }
+                    }
+                }
+            }
+        """),
+        {"argPath": path},
+    )["ldataResolvePathData"]
+
+    res: List[str] = []
+    for node in lrpd["childLdataTreeEdges"]["nodes"]:
+        res.append(node["child"]["name"])
+
+    return res
+
+
+class Team(TypedDict):
+    accountId: str
+
+
+class TeamMembersByUserIdNode(TypedDict):
+    team: Team
+
+
+class TeamMembersByUserId(TypedDict):
+    nodes: List[TeamMembersByUserIdNode]
+
+
+class TeamInfosByOwnerId(TypedDict):
+    nodes: List[Team]
+
+
+class UserInfoByAccountId(TypedDict):
+    defaultAccount: str
+    teamMembersByUserId: TeamMembersByUserId
+    teamInfosByOwnerId: TeamInfosByOwnerId
+
+
+class Bucket(TypedDict):
+    bucketName: str
+
+
+class LdataS3MountAccessProvensByGeneratedUsing(TypedDict):
+    nodes: List[Bucket]
+
+
+class LdataS3MountConfiguratorRolesByAccountIdNode(TypedDict):
+    ldataS3MountAccessProvensByGeneratedUsing: LdataS3MountAccessProvensByGeneratedUsing
+
+
+class LdataS3MountConfiguratorRolesByAccountId(TypedDict):
+    nodes: List[LdataS3MountConfiguratorRolesByAccountIdNode]
+
+
+class AccountInfoCurrent(TypedDict):
+    userInfoByAccountId: UserInfoByAccountId
+    ldataS3MountConfiguratorRolesByAccountId: LdataS3MountConfiguratorRolesByAccountId
+
+
+@cache
+def _get_all_valid_domains_for_account() -> List[str]:
```

Comment:
```suggestion
def _get_known_domains_for_account() -> List[str]:
```

e.g. `id.node` domains are valid but not in this list